### PR TITLE
Update Postgres Dependency in Helm charts

### PR DESCRIPTION
**Pull request by krithika369**
This PR aims to address the recent [CI error on main branch](https://github.com/caraml-dev/turing/actions/runs/3964075353/jobs/6792602502).

The Postgres chart being used no longer exists. Upgrading the dependency to the latest chart version `12.1.9` (which upgrades the Postgres image to `15.11.0-debian-11-r20` by default, but has been overridden in the values file, to `12.13.0`, primarily for the compatibility with the `migrate/migrate` image used by MLP to run the DB migrations : [ref](https://artifacthub.io/packages/helm/bitnami/postgresql?modal=values)).

Many of the existing values for Postgres configs are outdated with the new chart and accordingly, the default values and Helm docs are updated.

### CODE REVIEW

File: infra/charts/turing/subcharts/merlin/values.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 global:
   merlin: {}
+  mlp: {}
```
Comment (krithika369) [inline infra/charts/turing/subcharts/merlin/values.yaml:3]: Adding this for completeness. When installing the Turing chart, the master values file will define this, so it is not a problem. But if trying to dry-run just the Merlin chart, the absence of this field is problematic.

File: infra/charts/turing/values.yaml [hunk 2]
```diff
@@ -253,10 +253,6 @@ mlp:
   postgresql:
     # -- Name of MLP's Postgresql deployment
     nameOverride: postgresql-mlp
-    # -- Password for MLP Postgresql database
-    postgresqlPassword: mlp
```
Comment (krithika369) [inline infra/charts/turing/values.yaml:257]: This is now moved under a nested `auth` field and is anyway defined directly in the subchart with the same value. I believe it was added here as well as an example, can be cleaned up.

File: infra/charts/turing/values.yaml [hunk 3]
```diff
@@ -253,10 +253,6 @@ mlp:
...
-    tls:
```
Comment (krithika369) [inline infra/charts/turing/values.yaml:258]: This is already the default config now, removing.

File: .github/actions/setup-test-cluster/action.yaml [hunk 4]
```diff
@@ -193,6 +193,10 @@ runs:
         timeout: 600 # seconds
         max-restarts: -1
 
+    - name: Kubernetes namespace report
```
[the remark this answers is not in the document]
Comment (deadlycoconuts) [reply .github/actions/setup-test-cluster/action.yaml:196]: How did that report get generated in your example if you've used the `if: ${{ failure() }}` conditional? 😱

File: infra/charts/turing/subcharts/merlin/templates/_helpers.tpl [hunk 5]
```diff
@@ -27,10 +27,6 @@
 {{- printf "%s.%s.svc.cluster.local" (include "merlin.postgresql.name" .) .Release.Namespace -}}
 {{- end -}}
 
-{{- define "merlin.mlflow.backendStoreUri" -}}
-{{- printf "postgresql://%s:%s@%s:5432/%s" .Values.postgresql.postgresqlUsername .Values.postgresql.postgresqlPassword (include "merlin.db.host" .) .Values.postgresql.postgresqlDatabase -}}
-{{- end -}}
```
Comment (deadlycoconuts) [inline infra/charts/turing/subcharts/merlin/templates/_helpers.tpl:32]: Thanks for removing this!

**Review by deadlycoconuts — APPROVED** [review]
LGTM! Thanks a lot for updating this! 🚀

File: .github/actions/setup-test-cluster/action.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (krithika369) [reply .github/actions/setup-test-cluster/action.yaml:196]: Haha yes, that's right!

**Pull request merged**